src/main/webapp/assets/js/feedScript.js

```diff
@@ -605,16 +614,204 @@ function closeModal() {
  * @return {void}
  */
 async function submitModal() {
-  const collegeId = (new URLSearchParams(window.location.search)).get('collegeid');
+  // Disable multiple submissions.
+  submitModalButton.disabled = true;
 
-  // If one of the fields is empty, don't submit.
-  // Uses formElement.length - 1 to exclude the button element.
+  if (validateModal()) {
+    checkLocationAndSubmit();
+  } else {
+    submitModalButton.disabled = false;
+  }
+}
+
+/**
+ * Goes through the form elements and marks the invalid inputs.
+ * Returns whether all the inputs are valid.
+ * @return {boolean}
+ */
+function validateModal() {
+  const invalidIds = [];
+  const errorMessages = [];
   const formElements = modalForm.elements;
+
+  disableInjection(formElements);
+  validateModalText(invalidIds, errorMessages, formElements);
+  validateModalDate(invalidIds, errorMessages, formElements);
+  validateModalTime(invalidIds, errorMessages, formElements);
+  markInvalidInputs(invalidIds, errorMessages, formElements);
+
+  if (invalidIds.length === 0) {
+    return true;
+  } else {
+    return false;
+  }
+}
+
+/**
+ * Goes through the elements and replaces the tags with their HTML equivalent.
+ * @param {array} formElements
+ * @return {void}
+ */
+function disableInjection(formElements) {
   for (let i = 0; i < formElements.length - 1; i++) {
-    if (formElements[i].value.length == 0) {
-      return;
+    let eltValue = formElements[i].value;
+    eltValue = eltValue.replace(/</g, '&lt;').replace(/>/g, '&gt;');
+  }
+}
+
+/**
+ * Resets all the input backgrounds to modal grey and gets rid of previous error text.
+ * @param {array} formElements
+ * @return {void}
+ */
+function resetMarks(formElements) {
+  for (let i = 0; i < formElements.length - 1; i++) {
+    const elt = formElements[i];
```

Comment:
Same here. It will be difficult for someone who's reading this for the first time to know what elt means.

Reply:
Sounds good, I'll change it to element!